Sources/CoreMetrics/Metrics.swift

```diff
@@ -172,14 +172,19 @@ public class Gauge: Recorder {
     }
 }
 
+public enum TimeUnit {
+    case nanoSeconds, milliSeconds, seconds, minutes, hours, days
+}
```

Comment:
is there a non-foundation alternative to defining our own units?

/cc @weissi

Reply:
AFAIK no, so sadly currently all projects define their own time/interval etc types.
I think this is better than pulling in foundation for this

Reply:
Foundation has

- [the units and measurement stuff](https://developer.apple.com/documentation/foundation/unitduration) which has a bunch of that
- [TimeInterval](https://developer.apple.com/documentation/foundation/timeinterval) which is a `Double` of seconds so can express smaller amounts of time too

Reply:
I'd love to have a builtin type for this, but I could not find one in my searching.

Reply:
I guess the doubles argument also has fans... https://twitter.com/ID_AA_Carmack/status/418158611664097280 though not sure about it in the context of metrics.

Reply:
I think for now we can stick with this, since it does not require backends that always use NS to change from seconds to NS now.